src/Tests/Microsoft.Diagnostics.Monitoring.UnitTests/AuthenticationTests.cs

```diff
@@ -185,6 +185,72 @@ public async Task ApiKeyAuthenticationSchemeTest()
             Assert.NotNull(processes);
         }
 
+        /// <summary>
+        /// Tests that --temp-apikey flag can be used to generate a key.
+        /// </summary>
+        [Fact]
+        public async Task TempApiKeyTest()
+        {
+            await using MonitorRunner toolRunner = new(_outputHelper);
+            toolRunner.UseTempApiKey = true;
+
+            _outputHelper.WriteLine("Starting Monitor.");
```

Comment:
nit: These "Starting Monitor" logs shouldn't be necessary. There are a whole bunch of logs that already detail how dotnet-monitor is started and when in the test it occurs.